### ISSUE TITLE
Trim and interlink .md documentation

### ISSUE THREAD

**Issue by vemv** [threatgrid/ctim#321]
## Problem statement

As a developer, it can be somewhat difficult to learn about the models, even though all the data is there.

For example, checking the source of Casebook:

```clj
(def-entity-type Casebook
  {:description casebook-desc
   :reference "#"}
  c/base-entity-entries
  c/sourcable-object-entries
  c/describable-entity-entries
  (f/required-entries
   (f/entry :type CasebookTypeIdentifier))
  (f/optional-entries
   (f/entry :observables (f/seq-of c/Observable))
   (f/entry :bundle bun/Bundle)
   (f/entry :texts (f/seq-of Text))))
```

I see something pretty abstract and concise. I can certainly use IDE features to jump around etc, but it's easy to lose context after jumping for a while.

On the other hand, [Casebook.md](https://github.com/threatgrid/ctim/blob/375cc22e003dccd55615ddcfc72e1bf0f7530cd4/doc/structures/casebook.md), which is programatically generated, has kind of the opposite problem: all fields are expanded recursively, which results in a huge document to be rendered by the browser.

When facing such a large document, it can be difficult to discern the relevant parts, in which order should be the document read, etc.

tldr: I feel that one one should be able to see the models at a _balanced_ level of abstraction.

## Solution proposal

Refactor what is emitted to .md files / how.

I'd make every model expand recursively at (probably) 1 level at most. Any field that was nested beyond that would still be present, but as an hyperlink instead of as expanded text. The link would point to a sibling document in github.com (or the local filesystem, in a `dev` build).

By using links, we can trim a given page's size, while still not losing info.

This would have some similarities with JavaDoc - a reasonably successful system.
